lib/OpenQA/Client/Archive.pm

```diff
@@ -18,7 +18,8 @@ sub run ($self, $options) {
     my $res = $req->res;
     my $default_max_asset_size = 1024 * 1024 * 200;
     $options->{'asset-size-limit'} //= $default_max_asset_size;
-    $self->client->max_response_size($options->{'asset-size-limit'});
+    $options->{'no-limit'} //= 0;
+    $self->client->max_response_size($options->{all} ? 0 : $options->{'asset-size-limit'});
```

Comment:
Where is `$options->{all}` set? It doesn't show up on `openqa-cli archive -h` and it also doesn't look like this PR is introducing it. And wouldn't it make sense to let the user simply specify `--asset-size-limit 0` instead of having this special case?